src/Servers/HttpSys/src/NativeInterop/RequestQueue.cs

```diff
@@ -14,20 +15,54 @@ internal class RequestQueue
         private static readonly int BindingInfoSize =
             Marshal.SizeOf<HttpApiTypes.HTTP_BINDING_INFO>();
 
+        private readonly RequestQueueMode _mode;
         private readonly UrlGroup _urlGroup;
         private readonly ILogger _logger;
         private bool _disposed;
 
-        internal RequestQueue(UrlGroup urlGroup, string requestQueueName, ILogger logger)
+        internal RequestQueue(UrlGroup urlGroup, string requestQueueName, RequestQueueMode mode, ILogger logger)
         {
+            _mode = mode;
             _urlGroup = urlGroup;
             _logger = logger;
 
-            HttpRequestQueueV2Handle requestQueueHandle = null;
+            var flags = HttpApiTypes.HTTP_CREATE_REQUEST_QUEUE_FLAG.None;
+            Created = true;
+            if (_mode == RequestQueueMode.Attach)
+            {
+                flags = HttpApiTypes.HTTP_CREATE_REQUEST_QUEUE_FLAG.OpenExisting;
+                Created = false;
+            }
+
             var statusCode = HttpApi.HttpCreateRequestQueue(
-                    HttpApi.Version, requestQueueName, null, 0, out requestQueueHandle);
+                    HttpApi.Version,
+                    requestQueueName,
+                    null,
+                    flags,
+                    out var requestQueueHandle);
 
-            if (statusCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS)
+            if (_mode == RequestQueueMode.CreateOrAttach && statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_ALREADY_EXISTS)
+            {
+                // Tried to create, but it already exists so attach to it instead.
+                Created = false;
+                flags = HttpApiTypes.HTTP_CREATE_REQUEST_QUEUE_FLAG.OpenExisting;
+                statusCode = HttpApi.HttpCreateRequestQueue(
+                        HttpApi.Version,
+                        requestQueueName,
+                        null,
+                        flags,
+                        out requestQueueHandle);
+            }
+
+            if (flags == HttpApiTypes.HTTP_CREATE_REQUEST_QUEUE_FLAG.OpenExisting && statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_FILE_NOT_FOUND)
```

Comment:
Shouldn't this technically check CreateOrAttach and OpenExisting?

Reply:
That's why I'm checking the flags and not the enum.

Reply:
> For CreateOrAttach that should only happen with a strange race condition.

Yeah, the strange race condition is what I was referring to.

Reply:
I don't think it's really worth guarding against that race condition, since to do so would require some sort of looping.